bundler/lib/bundler/vendor/net-http-persistent/lib/net/http/persistent.rb

```diff
@@ -159,7 +159,14 @@ class Bundler::Persistent::Net::HTTP::Persistent
   # limits (typically windows).
 
   if Process.const_defined? :RLIMIT_NOFILE
-    DEFAULT_POOL_SIZE = Process.getrlimit(Process::RLIMIT_NOFILE).first / 4
+    open_file_limits = Process.getrlimit(Process::RLIMIT_NOFILE)
+
+    # Under Jruby on Windows Process responds to `getrlimit` but returns something that does not match docs
```

Comment:
The fix here is fine, if a little embarrassing. 😊 The bonus of this fix is that it will also solve the issue on non-native JRuby environments, like some enterprise web containers.

An alternative would be to use multiple assignment to get the first value. When it is a single value instead of an array it will still work.

```ruby
DEFAULT_POOL_SIZE, * = Process.getrlimit(Process::RLIMIT_NOFILE)
```

Oh, and it is spelled "JRuby" please ❤️

Reply:
Though, we do return a rather odd value so your original fix may be cleaner:

```
$ rvm jruby-9.2.14.0 do jruby -Xnative.enabled=false -e 'p Process.getrlimit(Process::RLIMIT_NOFILE)'
-e:1: warning: Process#getrlimit not supported on this platform
9223372036854775807
```

Reply:
Yeah, I had the same thought but I thought the value didn't make much sense as a resource limit so decided to disregard it.

> Oh, and it is spelled "JRuby" please heart

Sorry, I'll fix my typos in this regard :)